flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/table/catalog/hive/HiveCatalog.java

```diff
@@ -1734,6 +1814,152 @@ public CatalogColumnStatistics getPartitionColumnStatistics(
+        } catch (TException e) {
+            throw new CatalogException(
+                    String.format(
+                            "Failed to get table stats of table %s 's partitions %s",
+                            tablePath.getFullName(), String.valueOf(partitionSpecs)),
+                    e);
+        }
+
+        return result;
+    }
+
+    private CatalogColumnStatistics buildPartitionedByColumnSatistics(
+            ObjectPath tablePath, FieldSchema partitionedByColumn, List<String> partitionNames) {
+
+        Long nullCount = null;
+        Long ndv = HiveStatsUtil.max(0L, Long.valueOf(partitionNames.size()));
+
+        CatalogColumnStatisticDataBuilder<? extends CatalogColumnStatisticsDataBase> builder = null;
+
+        final CatalogColumnStatisticsDataLong.Builder longBuilder =
+                CatalogColumnStatisticsDataLong.builder();
+        longBuilder.ndv(ndv);
+        final CatalogColumnStatisticsDataDate.Builder dateBuilder =
+                CatalogColumnStatisticsDataDate.builder();
+        dateBuilder.ndv(ndv);
+        final CatalogColumnStatisticsDataDouble.Builder doubleBuilder =
+                CatalogColumnStatisticsDataDouble.builder();
+        doubleBuilder.ndv(ndv);
+        final CatalogColumnStatisticsDataString.Builder stringBuilder =
+                CatalogColumnStatisticsDataString.builder();
+        stringBuilder.ndv(ndv);
+        final CatalogColumnStatisticsDataBoolean.Builder booleanBuilder =
+                CatalogColumnStatisticsDataBoolean.builder();
+        final CatalogColumnStatisticsDataBinary.Builder binaryBuilder =
+                CatalogColumnStatisticsDataBinary.builder();
+
+        for (String partitionName : partitionNames) {
+            String[] partName = partitionName.split(Path.SEPARATOR);
+            String[] partVal = partName[partName.length - 1].split("=");
+            String value = partVal[1];
+            if (value.equals(HIVE_DEFAULT_PARTITION)) {
+                try {
+                    nullCount =
+                            getPartitionStatistics(
+                                            tablePath,
+                                            new CatalogPartitionSpec(
+                                                    Collections.singletonMap(
+                                                            partitionedByColumn.getName(),
+                                                            HIVE_DEFAULT_PARTITION)))
+                                    .getRowCount();
+                } catch (Exception e) {
+                    LOG.info(
+                            "Fail to get getPartitionColumnStatistics for partition {}={}",
+                            partitionedByColumn.getName(),
+                            HIVE_DEFAULT_PARTITION);
+                }
+            } else {
+                final String type = partitionedByColumn.getType();
+                if ("int".equalsIgnoreCase(type)) {
```

Comment:
Avoid to use "int" directly. May be we can do the check using LogicalType like `HiveStatsUtil#getColumnStatisticsData`